src/Relation/Binary/Apartness.agda

```diff
@@ -0,0 +1,68 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Apartness relations
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Relation.Binary.Apartness where
+
+open import Relation.Binary.Core using (Rel)
+open import Relation.Binary.Definitions using (Symmetric; Transitive; Reflexive; Irreflexive)
+open import Relation.Binary.Structures using (IsEquivalence)
+open import Level using (Level; _⊔_) renaming (suc to lsuc)
+open import Relation.Nullary using (¬_)
+open import Data.Sum using (_⊎_; [_,_]′)
+open import Data.Product using (_×_)
+
+
+module Properties {c ℓ₁} {Carrier : Set c} (_#_ : Rel Carrier ℓ₁) where
+
+  _¬#_ : Carrier → Carrier → Set _
+  x ¬# y = ¬ (x # y)
+
+  sym⇒sym-¬ : Symmetric _#_ → Symmetric _¬#_
+  sym⇒sym-¬ sym# x¬#y y#x = x¬#y (sym# y#x)
+
+  Comparison : Set _
+  Comparison = ∀ {x y z} → x # z → (x # y) ⊎ (y # z)
```

Comment:
I suspect that in order to make this usable you're going to have to make `y` explicit, as Agda won't be able to infer it. We also tend to name variables in the order that they're used, so it would be `∀ {x y} z → x # y → (x # z) ⊎ (z # y)`

Reply:
I've done this. It makes the logic in comp=>trans slightly more convoluted, but the original definition is indeed easier.